Sources/SourceKitLSP/SourceKitIndexDelegate.swift

```diff
@@ -24,27 +25,29 @@ actor SourceKitIndexDelegate: IndexDelegate {
   /// The count of pending unit events. Whenever this transitions to 0, it represents a time where
   /// the index finished processing known events. Of course, that may have already changed by the
   /// time we are notified.
-  let pendingUnitCount = AtomicInt32(initialValue: 0)
+  let pendingUnitCount = Atomic<Int32>(0)
 
   package init() {}
 
   nonisolated package func processingAddedPending(_ count: Int) {
-    pendingUnitCount.value += Int32(count)
+    pendingUnitCount.add(Int32(count), ordering: .acquiringAndReleasing)
   }
 
   nonisolated package func processingCompleted(_ count: Int) {
-    pendingUnitCount.value -= Int32(count)
-    if pendingUnitCount.value == 0 {
+    pendingUnitCount.subtract(Int32(count), ordering: .acquiringAndReleasing)
+    if pendingUnitCount.load(ordering: .acquiring) == 0 {
```

Comment:
Instead of having a `subtract` and `load`, we should probably use the value that is returned by `subtract` and then use it below as well. (having `load` be a method just surfaces this existing bug)

Reply:
> Should we just use sequentiallyConsistent as the ordering? As far as I can tell, the current releasing and acquiring produce the same semantics and I find sequentiallyConsistent easier to reason about. Or am I missing something.

I had issues with the tests before, but when running the tests now. The issue seems to have resolved itself.

> I think some of the Atomic<Int32> and Atomic<UInt32> can now become Atomic<Int> if that simplified their usage. They were only Int32 and UInt32 because of C interop.

At the start I didn't really wanna add too many changes, so I thought of keeping it as it was